ibm/resource_ibm_private_dns_custom_resolver_forwarding_rule.go

```diff
@@ -0,0 +1,197 @@
+				Description: "Descriptive text of the forwarding rule.",
+			},
+			pdnsCRFRType: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: InvokeValidator(pdnsCRForwardRule, "type"),
+				Description:  "Type of the forwarding rule.",
+			},
+			pdnsCRFRMatch: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The matching zone or hostname.",
+			},
+			pdnsCRFRForwardTo: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The upstream DNS servers will be forwarded to.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			pdnsCRFRRuleID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "the time when a forwarding rule ID is created, RFC3339 format.",
+			},
+		},
+	}
+}
+
+func resourceIBMPrivateDNSForwardingRuleValidator() *ResourceValidator {
+	validateSchema := make([]ValidateSchema, 1)
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 "type",
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Optional:                   true,
+			AllowedValues:              "hostname, zone",
+		},
+	)
+
+	resourceValidator := ResourceValidator{ResourceName: pdnsCRForwardRule, Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIbmDnsCrForwardingRuleCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	dnsSvcsClient, err := meta.(ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	instanceID := d.Get(pdnsInstanceID).(string)
+	resolverID := d.Get(pdnsCRFRResolverID).(string)
+	opt := dnsSvcsClient.NewCreateForwardingRuleOptions(instanceID, resolverID)
+
+	if des, ok := d.GetOk(pdnsCRFRDesctiption); ok {
+		opt.SetDescription(des.(string))
+	}
+	if t, ok := d.GetOk(pdnsCRFRType); ok {
+		opt.SetType(t.(string))
+	}
+	if m, ok := d.GetOk(pdnsCRFRMatch); ok {
+		opt.SetMatch(m.(string))
+	}
+	if _, ok := d.GetOk(pdnsCRFRForwardTo); ok {
+		opt.SetForwardTo(expandStringList(d.Get(pdnsCRFRForwardTo).([]interface{})))
+	}
+	result, _, err := dnsSvcsClient.CreateForwardingRuleWithContext(context, opt)
+
+	if err != nil || result == nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(convertCisToTfThreeVar(*result.ID, resolverID, instanceID))
+
+	return resourceIbmDnsCrForwardingRuleRead(context, d, meta)
+}
+
+func resourceIbmDnsCrForwardingRuleRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	dnsSvcsClient, err := meta.(ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	ruleID, resolverID, instanceID, err := convertTfToCisThreeVar(d.Id())
+	opt := dnsSvcsClient.NewGetForwardingRuleOptions(instanceID, resolverID, ruleID)
+	result, resp, err := dnsSvcsClient.GetForwardingRuleWithContext(context, opt)
+	if err != nil && resp.StatusCode == 404 {
```

Comment:
move inside the err block resp != nil then only check resp.StatusCode == 404
if err != nil  || result == nil { 
if resp != nil && resp.StatusCode == 404 {  
		d.SetId("")
		return nil
}
retrun dia.FromErr(fmt.Errorf("Error reading the forwarding rules %s:%s", err,resp)
}